### PR TITLE
CORE-8621 Refactor DiskResources module Part 1

### PR DESCRIPTION
**Pull request by ash3rz**
First of many, figured I should start separating these out before I end up with 1 gargantuan PR.

### CODE REVIEW

File: de-lib/src/main/java/org/iplantc/de/diskResource/client/views/metadata/dialogs/SelectMetadataTemplateDialog.java [hunk 1]
```diff
@@ -39,31 +41,23 @@
...
-        listStore = new ListStore<>(new ModelKeyProvider<MetadataTemplateInfo>() {
-            @Override
-            public String getKey(MetadataTemplateInfo item) {
-                return item.getId();
-            }
-        });
         this.appearance = appearance;
-        cm = buildColumnModel(showDownloadCell);
+        setModal(false);
+        setSize("400px", "400px");
```
[the remark this answers is not in the document]
Comment (ash3rz) [reply de-lib/src/main/java/org/iplantc/de/diskResource/client/views/metadata/dialogs/SelectMetadataTemplateDialog.java]: Ah, I missed one - thanks for catching!

**Comment by ash3rz** [comment]
Thanks for the review!

File: de-lib/src/main/java/org/iplantc/de/diskResource/client/presenters/callbacks/DeleteDataLinksCallback.java [hunk 2]
```diff
@@ -41,17 +39,15 @@ protected String getErrorMessageDefault() {
...
-    public void onSuccess(String result) {
-        JSONObject response = jsonUtil.getObject(result);
-        JSONArray tickets = jsonUtil.getArray(response, "tickets");
-
-        for (int i = 0; i < tickets.size(); i++) {
-            String ticketId = tickets.get(i).isString().toString().replace("\"", "");
+    public void onSuccess(UpdateTicketResponse result) {
+        List<String> tickets = result.getTickets();
+        tickets.forEach(ticket -> {
+            String ticketId = ticket.replace("\"", "");
```
Comment (psarando) [inline de-lib/src/main/java/org/iplantc/de/diskResource/client/presenters/callbacks/DeleteDataLinksCallback.java]: Nice refactor here (but also in general in this PR) 👍

File: de-lib/src/main/java/org/iplantc/de/diskResource/client/presenters/grid/GridViewPresenterImpl.java [hunk 3]
```diff
@@ -113,7 +110,7 @@ private SaveMetadataCallback(SaveAsDialog saveDialog) {
         public void onFailure(Throwable caught) {
             saveDialog.unmask();
             String fileName = saveDialog.getFileName();
-            if (caught.getMessage().contains("ERR_EXISTS")) {
+            if (caught.getMessage().contains(DiskResourceErrorCode.ERR_EXISTS.toString())) {
```
Comment (psarando) [inline de-lib/src/main/java/org/iplantc/de/diskResource/client/presenters/grid/GridViewPresenterImpl.java]: 👍

File: de-lib/src/main/java/org/iplantc/de/diskResource/client/presenters/grid/GridViewPresenterImpl.java [hunk 4]
```diff
@@ -665,15 +617,8 @@ public void onSuccess(FastMap<DiskResource> drMap) {
...
                                             ensureHandlers().fireEvent(new FetchDetailsCompleted());
-
-                                            // presenter.getView().unmaskDetailsPanel();
-                                            // presenter.unmaskVizMenuOptions();
                                         }
                                     });
-        // Need to mask the SendTo... options from the toolbar
-        // view.maskSendToCoGe();
-        // view.maskSendToEnsembl();
-        // view.maskSendToTreeViewer();
```
Comment (psarando) [inline de-lib/src/main/java/org/iplantc/de/diskResource/client/presenters/grid/GridViewPresenterImpl.java]: 👍

File: de-lib/src/main/java/org/iplantc/de/diskResource/client/views/dialogs/Md5DisplayDialog.java [hunk 5]
```diff
@@ -1,19 +1,40 @@
...
-                            int maxLength,
-                            String initialText,
-                            Validator<String> validator) {
-        super(caption, maxLength, initialText, validator);
-        setHeading(heading);
+    @Inject
+    public Md5DisplayDialog(GridView.Presenter.Appearance appearance) {
+        setHeading(appearance.md5Checksum());
+        setFieldLabelText(appearance.checksum());
+        setMaxLength(appearance.md5MaxLength());
```
Comment (psarando) [inline de-lib/src/main/java/org/iplantc/de/diskResource/client/views/dialogs/Md5DisplayDialog.java]: 👍

File: de-lib/src/main/java/org/iplantc/de/theme/base/client/diskResource/grid/GridViewDisplayStrings.properties [hunk 6]
```diff
@@ -19,7 +19,7 @@ shareByLinkFailure = Unable to generate your link. Please try again or contact s
 metadataOverwriteWarning = Metadata Template attributes on the selected file(s)/folder(s) will be overwritten. Do you wish to copy metadata from {0}?
 metadataManageFailure = Metadata management unavailable. Please try again or contact support.
 commentsManageFailure = Comments unavailable. Please try again or contact support.
-copyMetadata = Copy Metadata
+copyMetadata = Copy Metadata from: {0}
```
Comment (psarando) [inline de-lib/src/main/java/org/iplantc/de/theme/base/client/diskResource/grid/GridViewDisplayStrings.properties]: 👍

File: de-lib/src/main/java/org/iplantc/de/diskResource/client/events/DefaultUploadCompleteHandler.java [hunk 7]
```diff
@@ -88,69 +76,61 @@ public void onAfterCompletion() {
...
-    protected JSONObject buildPayload(String sourceUrl, String json) throws Exception {
-        JSONObject jsonObj = jsonUtil.getObject(jsonUtil.formatString(json));
-        if (jsonObj == null) {
-            throw new Exception(((DiskResourceView.Presenter.Appearance) GWT.create(DiskResourceView.Presenter.Appearance.class)).fileUploadsFailed(Lists.newArrayList(sourceUrl)) + ": " + json); //$NON-NLS-1$
-        }
-
+    Notification createNotification(File file, String sourceUrl) {
+        Notification notification = factory.notification().as();
+        PayloadData payload = factory.payloadData().as();
+        payload.setAction(PayloadData.ACTION_UPLOAD_COMPLETE);
```
Comment (psarando) [inline de-lib/src/main/java/org/iplantc/de/diskResource/client/events/DefaultUploadCompleteHandler.java]: 👍 🏆 ⭐️ ✨

**Comment by ash3rz** [comment]
Thanks! 😄

**Pull request merged**